app/dashboard/tasks.py

```diff
@@ -656,9 +658,15 @@ def calculate_trust_bonus(user_id, did, address):
 
                             if stamp_id_is_valid:
                                 # Save the stamp id and associate it with the Passport entry
-                                stamp_registry = PassportStamp.objects.update_or_create(user_id=user_id, stamp_id=stamp_id, defaults={
-                                    "passport": db_passport
-                                })
+                                stamp_registry = PassportStamp.objects.update_or_create(
+                                    user_id=user_id,
+                                    stamp_id=stamp_id,
+                                    stamp_provider=stamp_provider,
+                                    stamp_credential=stamp_credential,
```

Comment:
Does the `stamp_credential` and `stamp_provider` need to be in the kwargs? Or does it suffice to have it in the defaults ...
I am not sure if this impacts performance, but if we keep it in kwargs, we should make sure they are indexed?

Reply:
Ahh, good catch, yes this makes sense!

Reply:
resolved in 9b2008648f372ffb092e70b8b9e588ea2582a08e